### DISCUSSION TITLE
Conditional Element Redering

### DISCUSSION THREAD

**Discussion by s1n7ax** [leptos-rs/leptos#2595] [Q&A] [answered]
In React I can render a component based on a condition with Logical && Operator.

```jsx
function Mailbox(props) {
  const unreadMessages = props.unreadMessages;

  return (
    <div>
      <h1>Hello!</h1>

      { unreadMessages.length > 0 &&  
          <h2>
            You have {unreadMessages.length} unread messages.
          </h2>      
      }    

    </div>
  );
}
```

Is there similar option in Leptos? May be something like fragment so we can use it in else condition?

https://react.dev/reference/react/Fragment

**Comment by gbj** [comment, marked as answer]
Have you read the [Control Flow](https://book.leptos.dev/view/06_control_flow.html) chapter of the docs?

**Reply by s1n7ax** [reply]
For anyone looking for a code, 

Logo is option.

```
logo: Option<string>
```

`Option.map` can return be used to return Fn() -> Option<_> which can be used inside view directly.

```
let image = move || {
logo.map(|url| {
	view! {
		<Image
			width=100
			height=100
			class=styles::image
			quality=85
			blur=true
			src=url
			alt=logo_alt.unwrap_throw()
		/>
	}
})
};
```

**Reply by gbj** [reply]
In your case, remove the closure -- this is not reactive, and `Option<T>` implements `IntoView` if `T: IntoView`.
